### PR TITLE
DM-40474: Changes error handling to use IDDS

### PR DESCRIPTION
**Pull request by villarrealas**
This needs a practical test, but this PR should make the PanDA error checking about 10~30x faster than before, at the expense of an additional two libraries.

### CODE REVIEW

**Comment by eigerx** [comment]
I tried to use this branch to run `cm daemon` and `cm check` on the w_2023_34 DC2 run and got these errors:
```
cm daemon --fullname ${fullname} --max-running 30 --sleep-time 100
Traceback (most recent call last):
  File "/sdf/data/rubin/shared/campaigns/HSC-RC2/venvs/prod/bin/cm", line 5, in <module>
    from lsst.cm.tools.cli.commands import cli
  File "/sdf/data/rubin/shared/campaigns/DC2-test-med-1/cm_tools/src/lsst/cm/tools/cli/commands.py", line 6, in <module>
    from lsst.cm.tools.cli import options
  File "/sdf/data/rubin/shared/campaigns/DC2-test-med-1/cm_tools/src/lsst/cm/tools/cli/options.py", line 6, in <module>
    from click.decorators import F
ImportError: cannot import name 'F' from 'click.decorators' (/sdf/data/rubin/shared/campaigns/HSC-RC2/venvs/prod/lib/python3.10/site-packages/click/decorators.py)
```
```cm check --fullname ${fullname}
Traceback (most recent call last):
  File "/sdf/data/rubin/shared/campaigns/HSC-RC2/venvs/prod/bin/cm", line 5, in <module>
    from lsst.cm.tools.cli.commands import cli
  File "/sdf/data/rubin/shared/campaigns/DC2-test-med-1/cm_tools/src/lsst/cm/tools/cli/commands.py", line 6, in <module>
    from lsst.cm.tools.cli import options
  File "/sdf/data/rubin/shared/campaigns/DC2-test-med-1/cm_tools/src/lsst/cm/tools/cli/options.py", line 6, in <module>
    from click.decorators import F
ImportError: cannot import name 'F' from 'click.decorators' (/sdf/data/rubin/shared/campaigns/HSC-RC2/venvs/prod/lib/python3.10/site-packages/click/decorators.py)
```

File: src/lsst/cm/tools/core/panda_utils.py [hunk 1]
```diff
@@ -128,89 +110,66 @@ def get_errors_from_jeditaskid(dbi: DbInterface, jeditaskid: int):  # pragma: no
...
-            # because something went wrong
-    else:
+    if conn_status != 0:
         raise ValueError(f"Connection to Panda Failed with status {conn_status}")
 
-    # now we need to parse all the error codes for failed PandaIDs
     error_dicts = []
 
-    failed_jobs = [job for job in jobs_list if job.jobStatus == "failed"]
+    failed_jobs = [job for job in tasks if job["trans_exit_code"] != 0]
```
Comment (wguanicedew) [inline src/lsst/cm/tools/core/panda_utils.py:124]: For some jobs whose parent fails, the job will not be scheduled to run. Then the 'trans_exit_code' will be None. The job's status is 'Missing'.

File: src/lsst/cm/tools/core/panda_utils.py [hunk 2]
```diff
@@ -128,89 +110,66 @@ def get_errors_from_jeditaskid(dbi: DbInterface, jeditaskid: int):  # pragma: no
...
     if len(failed_jobs) == 0:
         return error_dicts
     else:
         for job in failed_jobs:
```
Comment (wguanicedew) [inline src/lsst/cm/tools/core/panda_utils.py:128]: If the job's status is 'Missing', all these values below will be None.  In this case, you can set the 'disgnostic_message' to something like 'Parent jobs failed, not schedule to run'.

File: src/lsst/cm/tools/core/panda_utils.py (hunk 1 above)
Comment (wguanicedew) [reply src/lsst/cm/tools/core/panda_utils.py:124]: failed_jobs = [job for job in tasks if job["trans_exit_code"] != 0 or job["trans_exit_code"] is None] ?

**Comment by villarrealas** [comment]
Adjusted for the changes that Wen Guan made and tested on the DC2 run, should be good to go.

File: src/lsst/cm/tools/core/panda_utils.py [hunk 3]
```diff
@@ -109,7 +84,7 @@ def get_jeditaskid_from_reqid(reqid: int, username: str) -> list[int]:  # pragma
 )
 
 
-def get_errors_from_jeditaskid(dbi: DbInterface, jeditaskid: int):  # pragma: no cover
+def get_errors_from_jeditaskid(dbi: DbInterface, conn, panda_reqid: int, jeditaskid: int):  # pragma: no cover
```
Comment (eigerx) [inline src/lsst/cm/tools/core/panda_utils.py:87]: I see `pragma: no cover` ; do we want to work on some unit tests?

File: src/lsst/cm/tools/core/panda_utils.py [hunk 4]
```diff
@@ -128,89 +110,73 @@ def get_errors_from_jeditaskid(dbi: DbInterface, jeditaskid: int):  # pragma: no
...
+                    error_dict["diagnostic_message"] = job["job_dispatcher_error_diag"]
+                elif job["sup_error_code"] != 0:
+                    error_dict["panda_err_code"] = "sup, " + str(job["sup_error_code"])
+                    error_dict["diagnostic_message"] = job["sup_error_diag"]
+                elif job["task_buffer_error_code"] != 0:
+                    error_dict["panda_err_code"] = "taskbuffer, " + str(job["task_buffer_error_code"])
+                    error_dict["diagnostic_message"] = job["task_buffer_error_diag"]
+                else:
+                    error_dict["panda_err_code"] = "unknown"
+                    error_dict["diagnostic_message"] = "check the logs"
```
Comment (eigerx) [inline src/lsst/cm/tools/core/panda_utils.py:170]: Nice logic!

**Review by eigerx — APPROVED** [review]

**Pull request merged**